sdk/python/lib/pulumi/resource.py

```diff
@@ -852,6 +853,7 @@ def __init__(
                resource.
         :param bool remote: True if this is a remote component resource.
         :param bool dependency: True if this is a synthetic resource used internally for dependency tracking.
+        :param Optional[Awaitable[Optional[str]]] package: The package reference for this resource.
```

Comment:
I'm surprised that this isn't a resource option, like `version` and `pluginDownloadURL`.

Reply:
It's not something I expect users to be able to edit, and I don't think it needs plumbing all the way through the transforms system.

Reply:
Also if it was a resource option it means for providers with large parameter values (imagine like a CRD provider with a dozen CRD yamls as it's parameter) that data needs to be sent over the wire over and over again. Simple but it was considered scary in cost.